tracking/src/main/java/org/hps/recon/tracking/TrackDataDriver.java

```diff
@@ -239,27 +247,58 @@ protected void process(EventHeader event) {
                     
                     // Get the HelicalTrackHit corresponding to the 
                     // RotatedHelicalTrackHit associated with a track
+                    hitToRotated.from(rotatedStereoHit);
+                    TrackerHit th = (TrackerHit) hitToRotated.from(rotatedStereoHit);
+                    if(th instanceof HelicalTrackHit)
+                    {
                     helicalTrackHit = (HelicalTrackHit) hitToRotated.from(rotatedStereoHit);
-                    ((HelicalTrackHit) rotatedStereoHit).setPosition(stereoHitPosition.v());
+                    }
+                    else
+                    {
+                        helicalTrackHit = TrackUtils.makeHelicalTrackHitFromTrackerHit(th);
+                    }
+                    (rsHit).setPosition(stereoHitPosition.v());
                     stereoHitPosition = CoordinateTransformations.transformVectorToDetector(stereoHitPosition);
                     helicalTrackHit.setPosition(stereoHitPosition.v());
 
                     // Loop over the clusters comprising the stereo hit
-                    for (HelicalTrackStrip cluster : ((HelicalTrackCross) rotatedStereoHit).getStrips()) {
-
-                        totalT0 += cluster.time();
+                    // this code appears to exist to provide the following:
+                    // totalT0
+                    // totalHits
+                    // trackerVolume (0 == top, 1 == bottom)
+                    //
+                    // Can we do this without resort to HelicalTrackCross?
+                    for (Object o : rotatedStereoHit.getRawHits())
+                    {
+                        RawTrackerHit rth = (RawTrackerHit) o;
+//                        System.out.println(rth);
+                        totalT0 += rth.getTime();
```

Comment:
Thanks for noticing this. I'll double-check on any differences in the totalT0 reported by both methods and modify if necessary.